seisbench/models/deepdenoiser.py

```diff
@@ -284,3 +283,454 @@ def forward(self, x, mid):
+        stride: Union[int, tuple[int, int]] = 1,
+        drop_rate: float = 0.3,
+        use_bias: bool = False,
+    ):
+        super().__init__()
+        self.block = nn.Sequential(
+            nn.ConvTranspose2d(
+                in_channels=in_channels,
+                out_channels=out_channels,
+                kernel_size=kernel_size,
+                stride=stride,
+                padding=1,
+                output_padding=1,
+                bias=use_bias,
+            ),
+            nn.BatchNorm2d(num_features=out_channels),
+            nn.ReLU(),
+            nn.Dropout(drop_rate),
+        )
+
+    def forward(self, x):
+        return self.block(x)
+
+
+class SeisDAE(DeepDenoiser):
+    """
+    Seismic Denoising Autoencoder using U-Net Architecture with additional attention gates.
+    A configurable denoising autoencoder for seismic waveform data that operates in the
+    time-frequency domain using the Short-Time Fourier Transform (STFT). The model is based
+    on a U-Net structure with optional attention gates and skip connections.
+
+    :param in_samples: Length of the input waveform in samples. Default is 3000 samples.
+    :param in_channels: Number of input channels (e.g., 2 for real and imaginary STFT components).
+                        Default are 2 channels.
+    :param sampling_rate: Sampling rate of the waveform data in Hz.
+                          Default sampling rate is 100 Hz.
+    :param filters_root: Number of filters in the first convolutional layer (doubles with depth).
+                         Default is 8.
+    :param depth: Number of encoding/decoding levels in the U-Net.
+                  Default is 6 for STFT
+    :param kernel_size: Kernel size for convolutional layers.
+                        Default is (3, 3).
+    :param strides: Stride size used for down/upsampling using Conv2D and transpose Conv2D layers.
+                    Default is (2, 2).
+    :param output_activation: Activation function applied to final output
+                              Default is Softmax.
+    :param drop_rate: Dropout rate used throughout the network.
+                      Default drop_rate is 0
+    :param use_bias: Whether to use bias in convolutional layers.
+                     Default is False.
+    :param norm: Type of normalization applied to traces ("peak" or "std").
+                 Default is "peak"
+    :param nfft: Length of the FFT used, if a zero padded FFT is desired for scipy STFT.
+                 If None, the FFT length is nperseg.
+    :param nperseg: Length of each segment for scipy STFT. Default is 60
+    :param attention: Whether to use attention gates in skip connections or not.
+                      Default is False
+    :param kwargs: Additional arguments passed to the base `WaveformModel`.
+    """
+
+    _annotate_args = WaveformModel._annotate_args.copy()
+    _annotate_args["overlap"] = (_annotate_args["overlap"][0], 1500)
```

Comment:
You can change it from `1500` to `0.5` to automatically use half the window length as overlap. I added this option recently.